src/classes/CRLP_RollupUI_SVC.cls

```diff
@@ -693,76 +693,84 @@ public with sharing class CRLP_RollupUI_SVC {
     }
 
     /*******************************************************************************************************
-     * @description Handle the save (insert/update) of a Rollup__mdt record
-     * @param rollupCMT Stringified structure of a CRLP_RollupCMT.Rollup wrapper class
+     * @description Handle the save (insert/update/delete) of a Rollup__mdt record
+     * @param jsonRollup Stringified structure of a CRLP_RollupCMT.Rollup wrapper class
      * @return {JobId}-{DeveloperName} - ASync Job Id - Developer name of the newly inserted or updated
      * record; or an error message. The {DeveloperName} value can be used by the JS to retrieve the record after
      * the insert (deployment) completes.
      */
     @AuraEnabled
-    public static String saveRollup(String rollupCMT) {
+    public static String saveRollup(String jsonRollup) {
         try {
             // For new records, the RecordName property will be null by default. The process of deploying the
             // CMT data will generate a unique RecordName to use for the new record.
-            CRLP_RollupCMT.Rollup rollupCMTObj = (CRLP_RollupCMT.Rollup) JSON.deserialize(rollupCMT, CRLP_RollupCMT.Rollup.class);
-
-            // Check for duplicate Target fields on active Rollups
+            CRLP_RollupCMT.Rollup cmtRollup = (CRLP_RollupCMT.Rollup) JSON.deserialize(jsonRollup, CRLP_RollupCMT.Rollup.class);
+            
             for (Rollup__mdt rlp : CRLP_Rollup_SEL.cachedRollups) {
```

Comment:
Similarly `isDuplicate()` can traverse through rollups and compare them to the `cmtRollup`:
```
if (cmtRollup.isDuplicate()) {
    throw new RollupException(Label.CRLP_DuplicateTargetField);
}
```

where `CRLP_RollupCMT` method:
```
/*******************************************************************************************************
* @description Determines if the target rollup is a duplicate
*/
public Boolean isDuplicate() {
    for (Rollup__mdt rlp : CRLP_Rollup_SEL.cachedRollups) {
        if (rlp.DeveloperName != this.recordName
            && rlp.Active__c
            && this.summaryObject == rlp.Summary_Object__r.QualifiedApiName
            && this.summaryField == rlp.Summary_Field__r.QualifiedApiName
            && !Test.isRunningTest()
        ) {
            return true;
        }
    }
    
    return false;
}
```